Source/NSDistributedLock.m

```diff
@@ -263,11 +263,23 @@ - (BOOL) tryLock
 	  attributes = [mgr fileAttributesAtPath: _lockPath
 				    traverseLink: YES];
 	  if (attributes == nil)
-	    {
-	      [NSException raise: NSGenericException
-		format: @"Unable to get attributes of lock file we made at %@",
-		_lockPath];
-	    }
+            {
+              /*
+               * We've seen cases where several instances of the same
+               * program were launched at once, where this exception
+               * got raised. It may be possible that one process removed
+               * the lock we've been using here, and with the right timing
+               * the exception could get raised, however we would only
+               * expect a failure in setting the lock:
+               */
+              /*
+               * [NSException raise: NSGenericException
+               *             format: @"Unable to get attributes of lock file we made"];
+               */
+              NSLog(@"Unable to get attributes of lock file we made");
+              return NO;
```

Comment:
This seems wrong to me at it would leave _localLock in the lock state. Also the above description is rather unlikely. We only end up in this branch, when we actually created the directory ourselves. I don't think another process would remove it in the meantime. Most likely there is a completely different issue here that this patch would just hide.

Reply:
Made a change here to unlock _localLock.  Aren't we leaving it in a lock state if we throw an exception as well?